### DISCUSSION TITLE
[v5] raw command to mysql

### DISCUSSION THREAD

**Discussion by rbartholomay** [adonisjs/core#1933] [Help]
Hi there,

i wan´t to run the command "SET foreign_key_checks = 0" to my dasebase, on a bulk opertation.

But it seems that is no executed on the database. Whats wrong?

```javascript
Database.raw('SET foreign_key_checks = 0;')
```

Bye!

**Comment by LeCoupa** [comment]
Have you tried with `await`?

```javascript
await Database.raw('SET foreign_key_checks = 0;')
```

This is using Knex.js behind the curtain. Here is the documentation for the [`raw()` method](http://knexjs.org/#Schema-raw).

You should also be able to run it inside one of your migrations in the `up()` or `down()` methods:

```javascript
this.schema.raw('SET foreign_key_checks = 0;')
```

**Reply by rbartholomay** [reply]
it seems that the problem is not the raw command. I use this inside transactions and it will complety run if i not using lucid models.

Below is my code and it runs fine. But if i use the lucid Models inside the transactions nothing will be saved...

Strange...

```javascript 
          await Database.transaction(async (trx) => {
            await trx.rawQuery('SET foreign_key_checks=?', [0])

            await trx.from(CollectorIdrCatalogItem.table)
              .where('target', target)
              .delete()

            await trx.from(CollectorIdrCatalogRelation.table)
              .where('target', target)
              .delete()

            await trx.table(CollectorIdrCatalogItem.table).multiInsert(data)
            await trx.table(CollectorIdrCatalogRelation.table).multiInsert(ccRelations)

            //await CollectorIdrCatalogItem.createMany(data, { client: trx }) // this will not run :-(
            //await CollectorIdrCatalogRelation.createMany(ccRelations, { client: trx }) // this will not run :-(

            await trx.rawQuery('SET foreign_key_checks=?', [1])
          })
```

**Reply by LeCoupa** [reply]
Does it work when you move the raw query outside of the transaction? Something like:

```javascript
await Database.raw('SET foreign_key_checks = 0;')
await Database.transaction(async (trx) => { … })
await Database.raw('SET foreign_key_checks = 1;')
```
